File: backend/internal/dtos/reading_preferences.go
package dtos

// ReadingPreferences represents the user's preferences for reading.
//
// Fields:
//   - AtomicReading (bool): If true, reading will happen atomically.
//   - Font (string): The preferred font for reading.
//   - Theme (string): The preferred theme for reading.
//   - TTS (TTSPreferences): The preferences for text-to-speech.
type ReadingPreferences struct {
	AtomicReading bool           `json:"atomicReading"`
	Font          string         `json:"font"`
	Theme         string         `json:"theme"`
	TTS           TTSPreferences `json:"tts"`
}

// TTSPreferences represents the user's preferences for text-to-speech.
//
// Fields:
//   - Autoplay (bool): Whether text-to-speech should automatically start playing.
//   - Voice (string): The name or identifier of the desired voice.
//   - Rate (int): The playback rate, typically a percentage of normal speed.
type TTSPreferences struct {
	Autoplay bool   `json:"autoplay"`
	Voice    string `json:"voice"`
	Rate     int    `json:"rate"`
}

File: frontend/schemas/ReadingPreferences.ts
import { z } from 'zod';

// Define the Zod schema for the TTS part of ReadingPreferences
const TtsSchema = z.object({
  autoplay: z.boolean().nullable().optional(),
  voice: z.string().nullable().optional(),
  rate: z.number().nullable().optional(),
});

// Define the Zod schema for ReadingPreferences
const ReadingPreferencesSchema = z.object({
  atomicReading: z.boolean().nullable().optional(),
  font: z.string().nullable().optional(),
  theme: z.string().nullable().optional(),
  tts: TtsSchema,
});

type Tts = z.infer<typeof TtsSchema>;
type ReadingPreferences = z.infer<typeof ReadingPreferencesSchema>;

export { type ReadingPreferences, type Tts, TtsSchema, ReadingPreferencesSchema };
